src/game/physics.cpp

```diff
@@ -1055,7 +1055,7 @@ namespace physics
             if(curmat == MAT_WATER || oldmat == MAT_WATER)
             {
                 const bvec &watercol = getwatercol((curmat == MAT_WATER ? matid : pl->inmaterial) & MATF_INDEX);
-                mattrig(bottom, watercol, 0.5f, int(radius), PHYSMILLIS, 0.25f, PART_SPARK, curmat != MAT_WATER ? S_SPLASH2 : S_SPLASH1);
+                mattrig(bottom, watercol, 0.5f, int(radius), PHYSMILLIS, 0.25f, PART_SPARK, (curmat != MAT_WATER ? S_SPLASH2 : S_SPLASH1));
```

Comment:
Just a readability change, right?

Reply:
Kind of: the compiler is probably interpreting this as `MAT_WATER ? S_SPLASH2 : S_SPLASH1` and then comparing against `curmat`

Reply:
!= is computed before ?:

Reply:
@robalni is right here. If you want to make this more explicit (and improve readability), you have to put brackets around the `!=` comparison.

Reply:
That does not work I'm afraid:
```
bn/src/game/physics.cpp: In function ‘void physics::updatematerial(physent*, const vec&, const vec&, bool)’:
bn/src/game/physics.cpp:1058:115: warning: ‘?:’ using integer constants in boolean context, the expression will always evaluate to ‘true’ [-Wint-in-bool-context]
 1058 |                 mattrig(bottom, watercol, 0.5f, int(radius), PHYSMILLIS, 0.25f, PART_SPARK, (curmat != MAT_WATER) ? S_SPLASH2 : S_SPLASH1);
      |                                                                                                                   ^
```

Reply:
I think the problem here is that in the definition of the macro `mattrig`, the last argument `mw` is used like:

```c
if(mw >= 0)
```

and this expands to:

```c
if(curmat != MAT_WATER ? S_SPLASH2 : S_SPLASH1 >= 0)
```

which is the same as:

```c
if( (curmat != MAT_WATER) ? S_SPLASH2 : (S_SPLASH1 >= 0) )
```

so I think this macro needs to be fixed by adding parentheses around the arguments:

```c
if((mw) >= 0)
```

Reply:
Awesome catch! Looks a bit ugly now though 👀 

Not even worth converting into a function because this macro is used twice in the codebase...

Reply:
In that case it might just be removed entirely. That tiny bit of code duplication is acceptable.